test/run_test.py

```diff
@@ -148,22 +152,30 @@ def run_test(executable, test_module, test_directory, options, *extra_unittest_a
 def test_cuda_primary_ctx(executable, test_module, test_directory, options):
     return run_test(executable, test_module, test_directory, options, '--subprocess')
 
-def test_cpp_extensions(executable, test_module, test_directory, options):
+
+def _test_cpp_extensions_aot(executable, test_module, test_directory, options, use_ninja):
+    if use_ninja:
+        try:
+            cpp_extension.verify_ninja_availability()
+        except RuntimeError:
+            print(CPP_EXTENSIONS_ERROR)
+            return 1
+
+    OLD_USE_NINJA = os.environ.get('USE_NINJA', '')
     try:
-        cpp_extension.verify_ninja_availability()
-    except RuntimeError:
-        print(CPP_EXTENSIONS_ERROR)
-        return 1
-    cpp_extensions_test_dir = os.path.join(test_directory, 'cpp_extensions')
-    return_code = shell([sys.executable, 'setup.py', 'install', '--root', './install'],
-                        cwd=cpp_extensions_test_dir)
-    if return_code != 0:
-        return return_code
-    if sys.platform != 'win32':
-        return_code = shell([sys.executable, 'setup.py', 'install', '--root', './install'],
-                            cwd=os.path.join(cpp_extensions_test_dir, 'no_python_abi_suffix_test'))
+        os.environ['USE_NINJA'] = str(1 if use_ninja else 0)
```

Comment:
I am not sure this is worth fixing, but I want to call out that this is not the idiomatic way to modify the environment for a subprocess call. More idiomatic is to explicitly pass in the desired environment to the subprocess call. This approach is thread safe where modifying the global environ is not.

Reply:
Thanks for calling it out -- the reason why I wrote it this way was because I didn't know one could pass an environment into subprocess. I've fixed it to pass the environment to the subprocess call instead of patching it